leopard8_test.go

```diff
@@ -0,0 +1,26 @@
+package reedsolomon
+
+import "testing"
+
+func TestAddMod8(t *testing.T) {
+	type testCase struct {
+		x        ffe8
+		y        ffe8
+		expected ffe8
+	}
+	testCases := []testCase{
+		{x: ffe8(1), y: ffe8(2), expected: ffe8(3)},
+		{x: ffe8(253), y: ffe8(1), expected: ffe8(254)},
+		{x: ffe8(254), y: ffe8(1), expected: ffe8(0)},
+		{x: ffe8(254), y: ffe8(2), expected: ffe8(1)},
+		{x: ffe8(255), y: ffe8(0), expected: ffe8(0)},
+		{x: ffe8(255), y: ffe8(1), expected: ffe8(1)},
+		// {x: ffe8(255), y: ffe8(255), expected: ffe8(0)},
```

Comment:
@klauspost I commented this test case out because it currently fails but I think it is a valid test case. I'm not that familiar with finite fields but I assumed this should return `0`. If you agree then I'll modify the implementation of `addMod8` and uncomment this test.

Reply:
Yes, that would align with my expectation as well.